corehq/apps/domain/decorators.py

```diff
@@ -241,6 +241,22 @@ def __inner__(request, domain, *args, **new_kwargs):
     return decorator
 
 
+def api_domain_view(view):
+    """
+    Decorate this with any domain view that should be accessed via api only
+    """
+    @wraps(view)
+    @api_key()
```

Comment:
I am tempted to add the `csrf_exempt`, but will let people who use this decorator add it to their view separately, and by default we will be safe than not